src/retrieve.js

```diff
@@ -0,0 +1,183 @@
+import MiniPromise from './ponyfills/minipromise.js';
+import base64 from './ponyfills/base64.js';
+import { assign} from './ponyfills/objects.js';
+
+const URL_PATTERN = /^([a-z]+):\/\/([a-z\-_\.]+)(\/.*)/i;
```

Comment:
Won't match URLs without a trailing slash, such as `https://google.com`, nor URLs with port numbers.

This regex is slightly more lenient in the hostname part but gets the job done without extra verbosity. The browser's `new URL()` and Node's `url.parse()` could be good options too.
```suggestion
const URL_PATTERN = /^([a-z]+):\/\/([^/]+)(.*)/i;
```

Reply:
Ah, good catch.